src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/EditingTargetCodeExecution.java

```diff
@@ -158,6 +161,44 @@ public void executeRange(Range range)
       executeRange(range, null, false);
    }
    
+   public void sendSelectionToTerminal(boolean terminalExecuteWhenNotFocused)
+   {
+      if (terminalExecuteWhenNotFocused && !docDisplay_.isFocused())
+      {
+         // the current editor may be in another window; if one of our source
+         // windows was last focused, use that one instead
+         SourceWindowManager manager = 
```

Comment:
Gotcha. However I'd argue that if you use the Code menu on the main window, it should act on the selection in that window. If I'm not mistaken that's how sending R code to the R console works (and we should probably behave symmetrically). For example this is where SendToConsole is dispatched:

https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/EditingTargetCodeExecution.java#L225-L229

It's a cross window event so when fired from a source window will get automatically routed to the main window:

https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/studio/client/workbench/views/console/events/SendToConsoleEvent.java#L23-L25